packages/medusa/src/interfaces/csv-parser.ts

```diff
@@ -0,0 +1,34 @@
+import { AwilixContainer } from "awilix"
+import { IParser } from "./abstract-parser"
+
+export type CsvParserContext = {
+  line: number
+  column: string
+}
+
+export interface IValidator<TLine> {
+  validate: (
+    value: string,
+    line: TLine,
+    context: CsvParserContext
+  ) => Promise<boolean>
+}
+
+export abstract class AbstractValidator<TLine> implements IValidator<TLine> {
+  constructor(protected readonly container: AwilixContainer) {}
+
+  abstract validate(value: string, line: TLine, context: CsvParserContext)
+}
+
+export type CsvSchema<TLine = any> = {
+  columns: {
```

Comment:
suggestion: I think that we should include a transformer to return the a value that could be the result of the input transformation. This transformer can be optional

Reply:
I agree. Should that replace the `mapTo` property? I feel like that also falls under the responsibility of a transformer.. wdyt?

Reply:
I don’t think so, the transformer only transform the input but can’t know on which column to map it

Reply:
alright! What do you think about the following naive version for the transformer:
```typescript
export type CsvSchema<TLine = unknown> = {
  columns: {
    name: string
    required?: boolean
    mapTo?: string
    validator?: AbstractCsvValidator<TLine>
    transformer?: (value: string, context: CsvParserContext<TLine>) => any
  }[]
}
```